lib/inputstreamhelper/__init__.py

```diff
@@ -137,21 +135,59 @@ def _load_widevine_config(cls):
         with open(cls._widevine_config_path(), 'r') as config_file:
             return loads(config_file.read())
 
+    @classmethod
+    def _has_vendor_widevine(cls):
+        """Whether the system has a vendor-supplied widevine CDM"""
+        widevine_cdm_filename = config.WIDEVINE_CDM_FILENAME[system_os()]
+        if widevine_cdm_filename is None:
+            return False
+
+        if cls._ia_cdm_path() is None:
+            return False
+
+        widevine_path = os.path.join(cls._ia_cdm_path(), widevine_cdm_filename)
+        vendor_widevine_path = '{0}_vendor{1}'.format(*os.path.splitext(widevine_path))
+        if not exists(vendor_widevine_path):
+            return False
+
+        return True
+
+    @classmethod
+    def _hardlink_vendor_widevine(cls):
+        """Hardlink vendor Widevine into place"""
+        widevine_cdm_filename = config.WIDEVINE_CDM_FILENAME[system_os()]
+        if widevine_cdm_filename is None:
+            return
+
+        if cls._ia_cdm_path() is None:
+            return
+
+        widevine_path = os.path.join(cls._ia_cdm_path(), widevine_cdm_filename)
```

Comment:
You could shorten lines 158-165 by using `_widevine_path()`.

Reply:
Yeah, I didn't like the interface, but I also did not want to rewrite too many things either as before you know it I am rewriting half of the add-on :-( Been there, done that.

Reply:
Alright, so here is that rewrite ;-) I had to split up `_widevine_path()` into two functions `_has_widevinecdm()` and `_widevinecdm_path()`. And did the same for `_has_vendor widevinecdm()` and `_vendor_widevinecdm_path()`.

Reply:
It is not exactly shorter though.